### PR TITLE
HTBHF-1178 Added children data of births to the response.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java [hunk 1]
```diff
@@ -49,10 +54,27 @@ public EligibilityResponse createEligibilityResponse(DWPEligibilityResponse dwpR
...
+
+        if (dwpChildren.size() >= hmrcChildren.size()) {
+            return dwpResponse.getChildren();
+        }
+
+        return hmrcResponse.getChildren();
+    }
+
+    // Return the list if it's not null, otherwise return any empty list.
+    private List<ChildDTO> getListOfEmptyIfNull(List<ChildDTO> children) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java]: `getListOfEmptyIfNull` is a bit of a mouthful. How about `getEmptyListIfNull`, or `notNullList`?

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java [hunk 2]
```diff
@@ -49,10 +54,27 @@ public EligibilityResponse createEligibilityResponse(DWPEligibilityResponse dwpR
...
+        return firstNonNull(children, ImmutableList.of());
+    }
+
     private Integer nullSafeMax(Integer count1, Integer count2) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java]: As discussed with David, there's a reason we're taking the maximum number of children from either response. When someone has a child (or adopts etc) they will move from CTC to UC, so in the cycle in which they transfer the CTC information will probably be out of date (having one fewer children) than UTC.

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java]: Renamed getEmptyListIfNull

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java (hunk 2 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java]: Yes, that's fine, is this documented somewhere? If not, can we make sure that it is?

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java [hunk 3]
```diff
@@ -49,10 +54,27 @@ public EligibilityResponse createEligibilityResponse(DWPEligibilityResponse dwpR
...
+        builder.children(createChildren(dwpResponse, hmrcResponse));
 
         return builder.build();
     }
 
+    private List<ChildDTO> createChildren(DWPEligibilityResponse dwpResponse, HMRCEligibilityResponse hmrcResponse) {
+        List<ChildDTO> dwpChildren = getEmptyListIfNull(dwpResponse.getChildren());
+        List<ChildDTO> hmrcChildren = getEmptyListIfNull(hmrcResponse.getChildren());
+
+        if (dwpChildren.size() >= hmrcChildren.size()) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java]: I wasn't aware of this choice, is this documented somewhere, seems an important thing to do with something like this?

Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java]: Updating the top level javadoc explaining the rules regarding children.

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java (hunk 2 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/eligibility/factory/EligibilityResponseFactory.java]: Updating the top level javadoc explaining the rules regarding children.

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/model/hmrc/HMRCEligibilityResponse.java [hunk 4]
```diff
@@ -24,4 +27,6 @@
     @JsonProperty("numberOfChildrenUnderFour")
     private final Integer numberOfChildrenUnderFour;
 
+    @JsonProperty("children")
```
[the remark this answers is not in the document]
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/eligibility/model/hmrc/HMRCEligibilityResponse.java]: As discussed we can leave these as separate classes for now.

**Pull request merged**